src/algmatch/stableMatchings/hospitalResidentsProblem/hrPreferenceInstance.py

```diff
@@ -19,4 +23,41 @@ def _load_from_file(self, filename: str) -> None:
     def _load_from_dictionary(self, dictionary: dict) -> None:
         reader = DictionaryReader(dictionary)
         self.residents = reader.residents
-        self.hospitals = reader.hospitals
\ No newline at end of file
+        self.hospitals = reader.hospitals
+
+    def check_preference_lists(self) -> None:
+        for r, r_prefs in self.residents.items():
+
+            if len(set(r_prefs["list"])) != len(r_prefs["list"]):
+                raise PrefRepError("resident",r)
+            
+            for h in r_prefs["list"]:
+                if h not in self.hospitals:
+                    raise PrefNotFoundError("resident",r,h)
+            
+        for h, h_prefs in self.hospitals.items():
+
+            if len(set(h_prefs["list"])) != len(h_prefs["list"]):
+                raise PrefRepError("hospital",h)
+            
+            for r in h_prefs["list"]:
+                if r not in self.residents:
+                    raise PrefNotFoundError("hospital",h,r)
+                
+    def clean_unacceptable_pairs(self) -> None:
+        for r, h in product(self.residents, self.hospitals):
+            if r not in self.hospitals[h]["list"] or h not in self.residents[r]["list"]:
+                try:
+                    self.residents[r]["list"].remove(h)
+                except ValueError:
+                    pass
+                try:
+                    self.residents[h]["list"].remove(r)

```

Comment:
this line raises a `KeyError` when I run `hrVerifyCorrectness.py`. If I change the except to `KeyError`, there is another error raised when actually constructing the stable matching. Not sure what the aim of this code is?

Reply:
I think that should say self.hospitals[h] and is just a typo.
ValueError is raised if you try to remove an element from an array which is not present in it.